pkg/virt-operator/resource/apply/admissionregistration.go

```diff
@@ -69,47 +63,37 @@ func (r *Reconciler) createOrUpdateValidatingWebhookConfiguration(webhook *admis
 		}
 	}
 
-	injectOperatorMetadata(kv, &webhook.ObjectMeta, version, imageRegistry, id, true)
 	if !exists {
-		r.expectations.ValidationWebhook.RaiseExpectations(kvkey, 1, 0)
-		_, err := r.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(context.Background(), webhook, metav1.CreateOptions{})
+		r.expectations.ValidationWebhook.RaiseExpectations(r.kvKey, 1, 0)
+		webhook, err := r.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(context.Background(), webhook, metav1.CreateOptions{})
 		if err != nil {
-			r.expectations.ValidationWebhook.LowerExpectations(kvkey, 1, 0)
+			r.expectations.ValidationWebhook.LowerExpectations(r.kvKey, 1, 0)
 			return fmt.Errorf("unable to create validatingwebhook %+v: %v", webhook, err)
 		}
+		SetValidatingWebhookConfigurationGeneration(&r.kv.Status.Generations, webhook)
 
 		return nil
 	}
 
-	if !objectMatchesVersion(&cachedWebhook.ObjectMeta, version, imageRegistry, id, kv.GetGeneration()) || !certsMatch {
-		// Patch if old version
-		var ops []string
+	modified := resourcemerge.BoolPtr(false)
+	existingCopy := cachedWebhook.DeepCopy()
+	expectedGeneration := ExpectedValidatingWebhookConfigurationGeneration(webhook, r.kv.Status.Generations)
 
-		// Add Labels and Annotations Patches
-		labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&webhook.ObjectMeta)
-		if err != nil {
-			return err
-		}
-		ops = append(ops, labelAnnotationPatch...)
-
-		// Add Spec Patch
-		webhooks, err := json.Marshal(webhook.Webhooks)
-		if err != nil {
-			return err
-		}
-		ops = append(ops, fmt.Sprintf(`{ "op": "replace", "path": "/webhooks", "value": %s }`, string(webhooks)))
-
-		_, err = r.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Patch(context.Background(), webhook.Name, types.JSONPatchType, generatePatchBytes(ops), metav1.PatchOptions{})
-		if err != nil {
-			return fmt.Errorf("unable to patch validatingwebhookconfiguration %+v: %v", webhook, err)
-		}
-
-		log.Log.V(2).Infof("validatingwebhoookconfiguration %v updated", webhook.GetName())
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, webhook.ObjectMeta)
+	// there was no change to metadata, the generation was right
+	if !*modified && existingCopy.ObjectMeta.Generation == expectedGeneration && certsMatch {
+		log.Log.V(4).Infof("validatingwebhookconfiguration %v is up-to-date", webhook.GetName())
 		return nil
+	}
 
+	webhook.ResourceVersion = cachedWebhook.ResourceVersion
+	webhook, err = r.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Update(context.Background(), webhook, metav1.UpdateOptions{})
```

Comment:
So in general, the advantage to `Patch` is that it's atomic. Sanity checks and update are performed in one step. `Update` by nature is prone to race conditions.

Also, it's usually a very good idea to use `Patch` when you don't "own" the type. In other words, `ValidatingWebhookConfigurations` was defined in somebody else's codebase. While it is true that Kubernetes is very careful about updating their API--it can happen. Where this can go very poorly is the way JSON marshalling is handled. Let's pretend that a field was added to the `ValidatingWebhookConfigurations` object (which can happen if Kubernetes is updated and KubeVirt isn't). If our client gets a copy of this object, it'll throw out this new field. When we update later, there's an inadvertent side effect--we effectively delete whatever value that field had in it.

So with that in mind, is there any way possible to use `Patch` here instead of `Update`?

Reply:
Switching this back to using `Patch` is defiantly possible if that is preferred. I'm not sure I understand the case you described however. Since we are using the `Patch` operation "replace" for the full `spec` / `workloads` key wouldn't Patch and Update react the same? Unless you are referring to a top level key being added/changed? 

`os = append(ops, fmt.Sprintf(`{ "op": "replace", "path": "/spec", "value": %s }`, string(newSpec)))`

Reply:
Spoke offline with @ashleyschuett to clarify. This comment is for the benefit of other reviewers. The trouble with Update is it replaces the entire record, while Patch only updates specific fields of interest.

When using "Update", if in the future, a new field is added to an object being reconciled, then our code would silently ignore it (read: delete the value). When performing an update on the entire record this would effectively delete that field--which isn't our intention. The danger of this would be potentially hot-looping.